HIP/hip-551.md

```diff
@@ -0,0 +1,89 @@
+## Abstract
+
+This HIP defines a mechanism to execute batch transactions such that a series of transactions(HAPI calls) depending on each other can be rolled into one transaction that passes the ACID test (atomicity, consistency, isolation, and durability)
+
+## Motivation
+
+The existing implementation of transactions in the Hedera network does not allow multiple different HAPI transactions to be called in one single network transaction that would have all the ACID properties.
+This makes it impossible to create more complicated flows without using smart contracts (which do not support all the HAPI transactions at this point) and listening to the mirror node to check the status of the previous transaction.
+This way we can also achieve an abstraction away from smart contracts
+
+## Rationale
+Atomic transactions chain is from the protobuf point of view just a list of `SignedTransactions` that's, why we based our new protobuf message on it,
+we only have to change the `bodyBytes` field to be `repeated`, that way we can store there multiple transaction bodies,
+`SignatureMap sigMap` can stay unchanged because it's defined as `repeated SignaturePair sigPair`.
+
+## User stories
+
+As a Hedera token service user, I want to be able to unfreeze an account, send an NFT, and freeze it again in one ACID transaction, that way I can achieve an account-bound NFT(nontransferable NFT) collection, without using the hedera smart contract service.
+We can use nontransferable NFTs e.g. as someone’s achievements, the creation of digital references, etc.
+
+As a Hedera token service user, I want to be able to unfreeze an account, send an FT and freeze it again in one ACID transaction, that way I can achieve an account-bound FT(nontransferable FT) collection, without using the hedera smart contract service.
+We can use nontransferable FT e.g. as game points, reputation points, etc.
+
+As a Hedera token service user, I want to be able to wipe a token, mint a token, and transfer a token in one ACID transaction.
+That way a backend developer could create complicated flows without the need to use smart contracts and the need to handle partially successful flows in the backend,
+e.g. the wipe transaction was successful but the mint transaction was not.
+
+As a Hedera service user, I want to be able to create batch transaction, that use multiple Hedera services e. g.
+I want to be able to wipe a token, mint a token, transfer a token, and submit a topic message in one ACID transaction.
+That way a backend developer could create complicated flows without the need to use smart contracts and the need to handle partially successful flows in the backend,
+e.g. the wipe transaction was successful but the mint transaction was not.
+
+## Specification
+We could create a new protobuf message based on `SignedTransaction` message:
+```
+message BatchTransaction {
+    /**
+    * TransactionBodies serialized into bytes, which must be signed
+    */
+    repeated bytes bodyBytes = 1;
```

Comment:
Q: do each of these transactions result in externalized individual transaction records in the record stream?
If individual records are exposed  is there a concept of parent and child transactions? If so you should supply an example of this.

Reply:
Instead, it might be possible to modify the `TransactionBody` protobuf message with a new field inside the `data` field that would be a repeated field of `TransactionBodyXXXX`  (now this is where it gets murky, should it be a repeated field of encoded transaction body XXX,  or should it be an explicitly defined new message that is a union of all the possible TransactionBodyXXXX types)  Either way, with an ordered list of TransactionBodyXXXX you can achieve the desired outcome and only have to generate the enclosing Transaction ID for the outer TransactionBody envelope.